### PR TITLE
Deal with invalid range passed to getSelectionIndexes

### PR DESCRIPTION
**Pull request by conorcussell**

### CODE REVIEW

File: packages/slate/src/models/node.js [hunk 1]
```diff
@@ -1405,6 +1405,11 @@ class Node {
       return null
     }
 
+    // if we've been given an invalid selection we can exit early.
+    if (!startKey && !endKey) {
```
Comment (ianstormtaylor) [inline packages/slate/src/models/node.js]: There's actually a `range.isUnset` property for checking this I believe, we should use that instead.

**Comment by conorcussell** [comment]
Fixed that! I've deliberately left it as a separate statement so it's clear its not just a performance issue, but it could easily just be part of the statement above with a different comment. I'll try and test it later today to check it doesn't break anything else.

**Comment by conorcussell** [comment]
This appears to fix both the cases from https://github.com/ianstormtaylor/slate/issues/1418

It makes the error from https://github.com/ianstormtaylor/slate/issues/1161 happen slightly later - it fails to focus, then when you click and start writing you get the error from assertKey, so its probably slightly worse in that case. I'll make a PR to make that error clearer anyway

**Comment by ianstormtaylor** [comment]
Thanks @conorcussell!

**Pull request merged**